models/DataObject/ClassDefinitionInterface.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\DataObject;
+
+use Pimcore\Model\DataObject\ClassDefinition\Data;
+use Pimcore\Model\DataObject\ClassDefinition\Data\FieldDefinitionEnrichmentModelInterface;
+use Pimcore\Model\DataObject\ClassDefinition\Helper\VarExportInterface;
+use Pimcore\Model\ModelInterface;
+
+interface ClassDefinitionInterface extends FieldDefinitionEnrichmentModelInterface, ModelInterface, VarExportInterface
+{
+    public static function getById(string $id, bool $force = false): ?ClassDefinitionInterface;
+
+    public static function getByName(string $name): ?ClassDefinitionInterface;
+
+    /**
+     * @param array<string, mixed> $values
+     */
+    public static function create(array $values = []): ClassDefinitionInterface;
+
+    /**
+     * @internal
+     */
+    public function rename(string $name): void;
+
+    /**
+     * @internal
+     */
+    public static function cleanupForExport(mixed &$data): void;
+
+    public function save(bool $saveDefinitionFile = true): void;
+
+    /**
+     * @internal
+     */
+    public function generateClassFiles(bool $generateDefinitionFile = true): void;
+
+    public function delete(): void;
+    /**
+     * @internal
+     */
+    public function isWritable(): bool;
+
+    /**
+     * @internal
+     */
+    public function getDefinitionFile(?string $name = null): string;
+
+    /**
+     * @internal
+     */
+    public function getPhpClassFile(): string;
+
+    /**
+     * @internal
+     */
+    public function getPhpListingClassFile(): string;
+
+    public function getId(): ?string;
+
+    public function getName(): ?string;
+
+    public function getCreationDate(): ?int;
+
+    public function getModificationDate(): ?int;
+
+    public function getUserOwner(): ?int;
+
+    public function getUserModification(): ?int;
+
+    public function setId(string $id): static;
```

Comment:
Can you add `@return $this` php docs to all methods which return $this?

Reply:
Does that have any benefit when using typehints? The ClassDefinition class doesn't have it either.

Reply:
Yes there are benefits, see comments in https://github.com/pimcore/pimcore/pull/13214